packages/bun-types/bun.d.ts

```diff
@@ -2982,6 +2982,10 @@ declare module "bun" {
        *
        */
       env?: Record<string, string | number>;
+      /**
+      * An AbortSignal to set spawn's signal.
+      */
+      signal?: AbortSignal | null;
```

Comment:
Thoughts on adding?

```ts
timeout?: number | null;
```

Which is basically short-hand for:

```ts
{
  signal: AbortSignal.timeout(...)
} 
```

And to avoid weird edge-cases of both being define, it's mutually exclusive.

Reply:
/cc @Jarred-Sumner @colinhacks 

Proposing this since `timeout` is an option in `child_process`.

Reply:
I like the signal property so people can just
```typescript
process.signal.addEventListener("abort",()=> ..);
``` 
and also you can check
`process.signal.aborted` to see if was aborted, so no need to change exitCode or an additional aborted: bool property

About timeout property, how It would behave in the case of the user informing the timeout + signal?, In `ClientRequest` we use our own `AbortController` and the signal passed from the user just call internally our `controller.abort()`, our timeout also just calls `controller.abort()` so the user can pass their signal an manually abort or wait for timeout, but it costs 1 extra instance of `AbortController`

Reply:
They'll confuse `signal` with process signals though

Reply:
Yeah people will confuse for sure, maybe the same strategy but instead of exposing `signal` only expose `aborted` with internally will be `signal.aborted`, in parameters case the norm still is `signal` with also can be confuse with process signals